### PR TITLE
WCP-1288 Changing simulator to static build and adding UDP video support 

### PR DESCRIPTION
**Pull request by m-karcz**

### CODE REVIEW

File: UdpVideo/UdpVideoController.cpp [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+void check(const char* dll)
+{
+    if(NULL == LoadLibraryEx(dll, NULL, LOAD_LIBRARY_SEARCH_USER_DIRS))
+    {
+        auto error = GetLastError();
+        std::cerr << "Failed to load: " << dll << " : " << error << std::endl;
+    }
+}
+
+struct UdpVideoController::Pimpl
```
Comment (rafalgliniak) [inline UdpVideo/UdpVideoController.cpp:6]: I would extract it to separate file, it's quite hard to read now.

File: UdpVideo/UdpVideoController.cpp [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+} //anon ns
+
+void check(const char* dll)
+{
+    if(NULL == LoadLibraryEx(dll, NULL, LOAD_LIBRARY_SEARCH_USER_DIRS))
+    {
+        auto error = GetLastError();
+        std::cerr << "Failed to load: " << dll << " : " << error << std::endl;
+    }
+}
```
Comment (rafalgliniak) [inline UdpVideo/UdpVideoController.cpp]: unused?

File: UdpVideo/UdpVideoController.cpp [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+        }
+    }
+
+    bool hasGstreamer()
+    {
+        return gstreamerHandle != NULL;
+    }
+    HMODULE WINAPI gstreamerHandle = NULL;
+    GstElement* gstPipeline = nullptr;
+    HMODULE loadLibrary(const char* dllName)
```
Comment (rafalgliniak) [inline UdpVideo/UdpVideoController.cpp]: I would consider creating some abstraction for this load/unload libraries with raii and also resolve. In general for now all the load lib stuff is mixed with parsing and launching the pipeline.

File: UdpVideo/UdpVideoController.cpp [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+    }
+};
+
+UdpVideoController::UdpVideoController(QObject* parent)
+    : QObject(parent)
+{
+    pimpl = std::make_unique<Pimpl>();
+}
+
+void UdpVideoController::startPipeline(const QString &beginning, const QString& ip, const QString& port)
```
Comment (rafalgliniak) [inline UdpVideo/UdpVideoController.cpp:55]: beginning?

File: UdpVideo/UdpVideoController.h [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+public:
+    explicit UdpVideoController(QObject* = nullptr);
+    ~UdpVideoController();
+
+    Q_INVOKABLE void startPipeline(const QString& beginning, const QString& address, const QString& port);
+    Q_INVOKABLE void stop();
+    Q_INVOKABLE bool hasGstreamer();
+private:
+    struct Pimpl;
+    std::unique_ptr<Pimpl> pimpl;
```
Comment (rafalgliniak) [inline UdpVideo/UdpVideoController.h:18]: could be
`std::unique_ptr<struct Pimpl> pimpl;`

**Comment by van9konst** [comment]
let me know when merge would be done.

File: UdpVideo/UdpVideoController.cpp (hunk 1 above)
Comment (m-karcz) [reply UdpVideo/UdpVideoController.cpp:6]: splitted

File: UdpVideo/UdpVideoController.cpp (hunk 3 above)
Comment (m-karcz) [reply UdpVideo/UdpVideoController.cpp]: splitted

File: DynamicGui/CMakeLists.txt [hunk 6]
```diff
@@ -4,6 +4,27 @@ set(SOURCES DynamicGuiController.cpp)
 add_library(DynamicGui ${HEADERS} ${SOURCES})
 add_library(DynamicGui::DynamicGui ALIAS DynamicGui)
 
-target_link_libraries(DynamicGui PUBLIC Qt5::Core zmq)
+target_link_libraries(DynamicGui PUBLIC
+                      Qt5::Core
+                      Qt5::Core
```
Comment (rafalgliniak) [inline DynamicGui/CMakeLists.txt]: .

**Review by rafalgliniak — DISMISSED** [review]

**Review by rafalgliniak — APPROVED** [review]

**Pull request merged**